internal/certs.go

```diff
@@ -79,19 +78,18 @@ func CreateServerTLSConfig(caCertPath, serverCertPath, serverKeyPath string) (*t
 		InsecureSkipVerify: true,
 		ClientCAs:          caCertPool,
 	}
-	tlsConfig.BuildNameToCertificate()
```

Comment:
Reading the deprecation notice this _seems_ to be fine, but I am not entirely sure since the notice is a bit vague.
Are you able to verify that the example server that uses this functionality still works fine? The clients should be able to connect if everything is working correctly.

Reply:
Sure, I can add a test with a TLS server.

Reply:
Thoughts on using https://github.com/open-telemetry/opamp-go/pull/202?

Reply:
Yeah #202 will need to be updated, it has conflicts. I don't think you need to add a test as part of this particular PR. If you can confirm by running the examples manually that they still work we should be good and can merge this PR.

Reply:
`make run-examples` should build and run the example server and agent and if you go to http://localhost:4321/ you should be able to see a connected agent. 

![image](https://github.com/open-telemetry/opamp-go/assets/4194920/fc7bd91b-0472-4134-b050-53c9077543ac)


Click on it, scroll down and ensure you see it connected using a client certificate, e.g.:

![image](https://github.com/open-telemetry/opamp-go/assets/4194920/53447e73-29ec-41e0-824c-75ad79ad1b93)

Reply:
Verified that the client certificate is shown on this page. Thanks for the instructions!

![image](https://github.com/open-telemetry/opamp-go/assets/1633772/f723127e-16cf-4d95-8e11-2a27a831e25b)